shared/src/components/activation/Activation.tsx

```diff
@@ -0,0 +1,91 @@
+import H from 'history'
+import { BehaviorSubject, Observable } from 'rxjs'
+import { first } from 'rxjs/operators'
+
+export interface ActivationStep {
+    id: string
+    title: string
+    detail: string
+    action: (h: H.History) => void
+}
+
+/**
+ * Tracks the activation status of the current user.
+ */
+export class ActivationStatus {
+    private steps_: ActivationStep[]
+    private fetchCompleted: () => Observable<{ [key: string]: boolean }>
+    private completed_: BehaviorSubject<{ [key: string]: boolean } | null>
+
+    /**
+     * Accepts the activation steps to track and a function that fetches the completed
+     * activation steps (in the form of a function that returns an Observable).
+     */
+    constructor(steps: ActivationStep[], fetchCompleted: () => Observable<{ [key: string]: boolean }>) {
```

Comment:
```suggestion
    constructor(private steps: ActivationStep[], private fetchCompleted: () => Observable<{ [key: string]: boolean }>) {
```

This allows you to remove the declarations `steps_` and `fetchCompleted_` above.